packages/next/src/server/stream-utils/node-web-streams-helper.test.ts

```diff
@@ -181,7 +182,30 @@ describe('node-web-stream-helpers', () => {
   describe('createBufferedTransformStream', () => {})
   describe('createInsertedHTMLStream', () => {})
   describe('renderToInitialFizzStream', () => {})
-  describe('createHeadInsertionTransformStream', () => {})
+  describe('createHeadInsertionTransformStream', () => {
+    it('should insert text into head tag', async () => {
+      const encoder = new TextEncoder()
+      function insert() {
+        return Promise.resolve('<foo>bar</foo>')
+      }
+      const input = new ReadableStream({
+        start(controller) {
+          controller.enqueue(
+            encoder.encode('<head><fuzz>buzz</fuzz></head><body></body>')
+          )
+          controller.close()
+        },
+      })
+      const stream = createHeadInsertionTransformStream(insert)
+      const output = input.pipeThrough(stream)
+      await processReadableStream(output, [
+        encoder.encode(
+          '<head><fuzz>buzz</fuzz><foo>bar</foo></head><body></body>'
+        ),
+        encoder.encode('<foo>bar</foo>'),
```

Comment:
Why does this method also emit the insertion value after the content? Is this intended? Am I setting it up correctly?

Reply:
I think I figured it out, `insert` is meant to return text only once (like a generator perhaps). This isn't _great_ but it works in this context.

Reply:
fixed in latest commit, but is still not a great DX